test/test_mat_mul_precision.py

```diff
@@ -0,0 +1,108 @@
+"""Numeric tests for default precision of mat mul.
+
+There are three similar test files, suffixed default, high, and highest.
+Unfortunately, the precision cannot reliably
+be dynamically changed between tensor operations, so
+the tests are split into three files to ensure a fresh
+environment for each test.
+"""
+
+import unittest
+import time
+
+import torch
+import torch_xla
+import torch_xla.backends
+
+import test_utils
+
+
+class TestMatMulPrecision(unittest.TestCase):
+
+  def _make_input(self):
+    eye = torch.eye(1024, device='cpu', dtype=torch.float64)
+    rand_ = torch.testing.make_tensor((1024, 1024),
+                                      dtype=torch.float64,
+                                      device="cpu",
+                                      low=0.99,
+                                      high=1.01)
+    return eye * rand_
+
+  # TODO: Figure out why either PT/XLA or unittest
+  # is unable to successfully run this test in a parameterized way.
```

Comment:
got it, I'm a bit surprised that tests below work and this one doesn't.
I had similar problem on GPU tests, the solution was to run them sequentially as you are doing here. 
My understanding that pytests is running tests in parallel, and therefore if torch_xla.backends.set_mat_mul_precision(precision) a global operation, meaning it affects everything that is executed on that process, tests running in parallel could interfere with each other, and if so I'd expect problems with individual tests test_highest/test_high/test_default.
Anyways, it is fine if it works this way.